tools/topology/topology2/include/components/smart_amp.conf

```diff
@@ -0,0 +1,137 @@
+#
+# Common widget smart_amp
+#
+# A generic smart_amp widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.smart_amp.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.smart_amp."N" {
+# 		index		1
+# 		type		"effect"
+#	}
+#
+# Where N is the unique instance number for the smart_amp object within the same alsaconf node.
+
+# set the default the SMART_AMP_UUID to be that of the test smart amp module.
+# The top-level machine topology must override this definition to use a different module
+Define {
+	SMART_AMP_UUID	"1e:96:7a:16:e4:8a:ea:11:89:f1:00:0c:29:ce:16:35"
+}
+
+Class.Widget."smart_amp" {
+	#
+	# Pipeline ID for the smart_amp widget object
+	#
+	DefineAttribute."index" {}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#
+	# smart_amp widget UUID
+	#
+	DefineAttribute."uuid" {
+		type	"string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."is_pages" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	Object.Control.bytes."1" {
+		!access  [
+			tlv_read
+			tlv_callback
+		]
+		Object.Base.extops.1 {
+			name	"extctl"
+			get 258
+			put 0
+		}
+		max 4096
+		Object.Base.data. 1 {
+			IncludeByKey.SMART_AMP_UUID {
+				# smart amp init data for the test smart amp module
+				"1e:96:7a:16:e4:8a:ea:11:89:f1:00:0c:29:ce:16:35"	{
+					bytes "0x53, 0x4f, 0x46, 0x34, 0x02, 0x00, 0x00, 0x00,
+					0x18, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08,
+					0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+					0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+					0x18, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00,
```

Comment:
What I would like to have is a line brake above this with a comment:
`				# smart amp payload`
Above the comment we have the ABI header, it is just easier to see

Reply:
@ujfalusi I cannot add that comment here as the data is a string.